bolero/optimizer/reps.py

```diff
@@ -0,0 +1,204 @@
+        Number of samples that will be used to update a policy.
+
+    bounds : array-like, shape (n_samples, 2), optional (default: None)
+        Upper and lower bounds for each parameter.
+
+    log_to_file : optional, boolean or string (default: False)
+        Log results to given file, it will be located in the $BL_LOG_PATH
+
+    log_to_stdout : optional, boolean (default: False)
+        Log to standard output
+
+    random_state : optional, int
+        Seed for the random number generator.
+    """
+    def __init__(self, initial_params=None, variance=1.0, covariance=None,
+                 epsilon=2.0, min_eta=1e-8, train_freq=25,
+                 n_samples_per_update=100, bounds=None, log_to_file=False,
+                 log_to_stdout=False, random_state=None):
+        self.initial_params = initial_params
+        self.variance = variance
+        self.covariance = covariance
+        self.epsilon = epsilon
+        self.min_eta = min_eta
+        self.train_freq = train_freq
+        self.n_samples_per_update = n_samples_per_update
+        self.bounds = bounds
+        self.log_to_file = log_to_file
+        self.log_to_stdout = log_to_stdout
+        self.random_state = random_state
+
+    def init(self, n_params):
+        """Initialize optimizer.
+
+        Parameters
+        ----------
+        n_params : int
+            dimension of the parameter vector
+        """
+        self.logger = get_logger(self, self.log_to_file, self.log_to_stdout)
+
+        self.random_state = check_random_state(self.random_state)
+
+        self.it = 0
+
+        if self.initial_params is None:
+            self.initial_params = np.zeros(n_params)
+        else:
+            self.initial_params = np.asarray(self.initial_params).astype(
+                np.float64, copy=True)
+        if n_params != len(self.initial_params):
+            raise ValueError("Number of dimensions (%d) does not match "
+                             "number of initial parameters (%d)."
+                             % (n_params, len(self.initial_params)))
+
+        self.params = None
+        self.reward = None
+
+        scaling = Scaling(variance=self.variance, covariance=self.covariance,
+                          compute_inverse=True)
+        self.policy_ = BoundedScalingPolicy(ConstantGaussianPolicy(
+            n_params, mean=scaling.inv_scale(self.initial_params),
+            random_state=self.random_state), scaling=scaling,
+            bounds=self.bounds)
+
+        # Maximum return obtained
+        self.max_return = -np.inf
+        # Best parameters found so far
+        self.best_params = self.initial_params.copy()
+
+        self.history_theta = deque(maxlen=self.n_samples_per_update)
+        self.history_R = deque(maxlen=self.n_samples_per_update)
+
+    def get_next_parameters(self, params, explore=True):
+        """Return parameter vector that shall be evaluated next.
+
+        Parameters
+        ----------
+        params : array-like
```

Comment:
the array probably needs to have a specific shape; I suggest adding ", shape = (num_params,)"